Sources/ToucanSDK/Toucan.swift

```diff
@@ -184,11 +167,27 @@ public struct Toucan {
                     .appending(path: result.destination.file)
                     .appendingPathExtension(result.destination.ext)
 
-                try result.contents.write(
-                    to: outputUrl,
-                    atomically: true,
-                    encoding: .utf8
-                )
+                switch result.source {
+                case .asset(let path, let contents):
+                    if let contents {
```

Comment:
Maybe we could add another case to handle different scenarios.
```
struct PipelineResult {
    enum Action {
        case content(contents: String)
        case asset(from: URL)
        case modifiedAsset(contents String)
    }

    var destination: Destination
    var action: Action
```

Reply:
Ok, `assetFile(path)` & `asset(contents)` is going to be it.